contracts/EscrowedNori.sol

```diff
@@ -0,0 +1,725 @@
+   * tokens to this contract will trigger tokensReceived as part of the lifecycle of the BridgedPolygonNORI transaction
+   */
+  IERC1820RegistryUpgradeable private _erc1820;
+
+  /**
+   * @notice Emitted on successful creation of a new escrow schedule.
+   */
+  event EscrowScheduleCreated(
+    address indexed recipient,
+    uint256 indexed startTime
+  );
+
+  /**
+   * @notice Emitted on when unreleased tokens of an active escrow schedule are revoked.
+   */
+  event UnreleasedTokensRevoked(
+    uint256 indexed atTime,
+    uint256 indexed removalId,
+    uint256 indexed escrowScheduleId,
+    uint256 quantity
+  );
+
+  /**
+   * @notice Emitted on withdwal of fully unlocked tokens.
+   */
+  event TokensClaimed(
+    address indexed from,
+    address indexed to,
+    uint256 quantity
+  );
+
+  // todo document expected initialzation state
+  // todo I switched the visibility of this function from public to external...
+  // is that right for an initializer or was there a reason it was public?
+  function initialize(
+    BridgedPolygonNORI bridgedPolygonNoriAddress,
+    Removal removalAddress
+  ) external initializer {
+    address[] memory operators = new address[](1);
+    operators[0] = _msgSender();
+    __Context_init_unchained();
+    __ERC165_init_unchained();
+    __AccessControl_init_unchained();
+    __AccessControlEnumerable_init_unchained();
+    __Pausable_init_unchained();
+    __ERC777PresetPausablePermissioned_init_unchained();
+    __ERC777_init_unchained("Escrowed NORI", "eNORI", operators);
+    _bridgedPolygonNori = bridgedPolygonNoriAddress;
+    _removal = removalAddress;
+    _ERC1820_REGISTRY.setInterfaceImplementer(
+      address(this),
+      ERC777_TOKENS_RECIPIENT_HASH,
+      address(this)
+    );
+    _grantRole(ESCROW_CREATOR_ROLE, _msgSender());
+  }
+
+  /**
+   * @notice This function is triggered when BridgedPolygonNORI is sent to this contract
+   *
+   * @dev Sending BridgedPolygonNORI to this contract triggers the tokensReceived hook defined by the ERC-777 standard
+   * because this contract is a registered ERC777 tokens recipient.
+   *
+   * [See here for more](
+   * https://github.com/ethereum/EIPs/blob/master/EIPS/eip-777.md#erc777tokensrecipient-and-the-tokensreceived-hook)
+   */
+  function tokensReceived(
+    address sender,
+    address,
+    address,
+    uint256 amount,
+    bytes calldata userData,
+    bytes calldata operatorData
+  ) external override {
+    require(
+      _msgSender() == address(_bridgedPolygonNori),
+      "eNORI: not BridgedPolygonNORI"
+    );
+    require(
+      hasRole(ESCROW_CREATOR_ROLE, sender),
+      "eNORI: sender is missing role ESCROW_CREATOR_ROLE"
+    );
+    uint256 removalId = abi.decode(userData, (uint256));
+    require(
+      removalId.supplierAddress() != address(0),
+      "eNORI: token send missing required userData"
+    );
```

Comment:
realizing that this same check happens in `ERC777Upgradeable._mint` which is called by the line below this where we invoke `_depositFor`.